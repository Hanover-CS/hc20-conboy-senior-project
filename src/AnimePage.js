/* Author: Patrick Conboy
 * Date Created: 1/5/2020
 * Description: This file creats the dynamic page for each 
 * individual anime in the database. On this page you see all
 * the information about the specific anime that is stored
 * in the database.
 */

import React, { Component } from 'react'

class AnimePage extends Component {
  constructor(props) {
    super(props)
    this.state = { animeName: "haikyuu" }
  }

  render() {
    return (
      <h3>this.state.animeName</h3>
    )
  }
}

export default AnimePage;