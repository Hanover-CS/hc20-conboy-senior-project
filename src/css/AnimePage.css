.addToListBtn:hover {
  cursor: pointer;
}

.removeFromListBtn:hover {
  cursor: pointer;
}