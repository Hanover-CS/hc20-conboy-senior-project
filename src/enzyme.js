/* Author: Patrick Conboy
 * Date Created: 1/1/2019
 * Description: This file initializes Enzyme for testing purposes.
 */

import Enzyme, { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
export { shallow, mount, render };
export default Enzyme;