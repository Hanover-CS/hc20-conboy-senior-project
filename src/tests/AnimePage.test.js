import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow, mount, render } from '../enzyme'

import AnimePage from '../AnimePage'

describe('AnimePage Tests', () => {
  test('anime page renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<AnimePage />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('anime page renders correctly', () => {
    const wrapper = shallow(<AnimePage />)

    expect(wrapper.find('#AnimePage')).toBeDefined()
  })

  test('snapshot of anime page hasn\'t changed', () => {
    const tree = renderer.create(<AnimePage />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})