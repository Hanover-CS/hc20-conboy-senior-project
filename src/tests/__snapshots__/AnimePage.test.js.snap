// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimePage Tests snapshot of anime page hasn't changed 1`] = `
<div
  className="animeData"
>
  <h2 />
  <h3>
    Synopsis: 
    
  </h3>
  <h3>
    Background: 
    
  </h3>
  <h3>
    Type: 
    
  </h3>
  <h3>
    Episodes: 
  </h3>
  <h3>
    Status: 
    
  </h3>
  <h3>
    Aired: 
    
  </h3>
  <h3>
    Premiered: 
    
  </h3>
  <h3>
    Producers: 
    
  </h3>
  <h3>
    Licensors: 
    
  </h3>
  <h3>
    Studios: 
    
  </h3>
  <h3>
    Source: 
    
  </h3>
  <h3>
    Genres: 
    
  </h3>
  <h3>
    Duration: 
    
  </h3>
  <h3>
    Rating: 
    
  </h3>
</div>
`;
